diff --git a/data/macros.meson b/data/macros.meson
index c5b90de03..483d1c5ad 100644
--- a/data/macros.meson
+++ b/data/macros.meson
@@ -1,3 +1,5 @@
+%_vpath_srcdir .
+%_vpath_builddir _build
 %__meson %{_bindir}/meson
 %__meson_wrap_mode nodownload
 %__meson_auto_features enabled
@@ -9,7 +11,10 @@
         if [ "$MESON_BUILD_NCPUS" -gt 1 ]; then echo "--num-processes $MESON_BUILD_NCPUS"; fi)
 
 %meson \
-    %set_build_flags \
+    export CFLAGS="${CFLAGS:-%__global_cflags}"       \
+    export CXXFLAGS="${CXXFLAGS:-%__global_cxxflags}" \
+    export FFLAGS="${FFLAGS:-%__global_fflags}"       \
+    export FCFLAGS="${FCFLAGS:-%__global_fcflags}"    \
     %{shrink:%{__meson} \
         --buildtype=plain \
         --prefix=%{_prefix} \
