diff --git a/data/macros.meson b/data/macros.meson
index b31f77e..8b71332 100644
--- a/data/macros.meson
+++ b/data/macros.meson
@@ -2,11 +2,12 @@
 %__meson_wrap_mode nodownload
 
 %meson \
-    export CFLAGS="${CFLAGS:-%__global_cflags}"       \
-    export CXXFLAGS="${CXXFLAGS:-%__global_cxxflags}" \
-    export FFLAGS="${FFLAGS:-%__global_fflags}"       \
-    export FCFLAGS="${FCFLAGS:-%__global_fcflags}"    \
-    export LDFLAGS="${LDFLAGS:-%__global_ldflags}"    \
+    export LANG=C.UTF-8                      \
+    export CFLAGS="${CFLAGS:-%optflags}"       \
+    export CXXFLAGS="${CXXFLAGS:-%optflags}" \
+    export FFLAGS="${FFLAGS:-%optflags}"       \
+    export FCFLAGS="${FCFLAGS:-%optflags}"    \
+    export LDFLAGS="${LDFLAGS:-%optflags}"    \
     %{__meson}                              \\\
         --buildtype=plain                   \\\
         --prefix=%{_prefix}                 \\\
@@ -23,17 +24,17 @@
         --localstatedir=%{_localstatedir}   \\\
         --sharedstatedir=%{_sharedstatedir} \\\
         --wrap-mode=%{__meson_wrap_mode}    \\\
-        %{_vpath_srcdir} %{_vpath_builddir} \\\
+        _build . \\\
         %{nil}
 
 %meson_build \
-    %ninja_build -C %{_vpath_builddir}
+    %ninja_build -C _build
 
 %meson_install \
-    %ninja_install -C %{_vpath_builddir}
+    %ninja_install -C _build
 
 %meson_test \
-    %ninja_test -C %{_vpath_builddir} || \
+    %ninja_test -C _build || \
     { rc=$?; \
       echo "-----BEGIN TESTLOG-----"; \
       cat %{_vpath_builddir}/meson-logs/testlog.txt; \
